android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Worker.kt

```diff
@@ -15,25 +15,35 @@
  */
 package com.uber.rib.core
 
+import kotlin.coroutines.CoroutineContext
 import kotlinx.coroutines.CoroutineDispatcher
 
 /**
  * Interface used when creating a manager or helper class that should be bound to an interactor's
  * lifecycle using a binder like [WorkerBinder]. The worker event is decoupled from the interactor's
  * actual lifecycle so that we're not stuck moving these classes around if there are other
  * lifecycles we're interested in.
+ *
+ * By default a [Worker] will be bound on the [CoroutineDispatcher] defined at [WorkerBinder.bind]
+ * call, except when the [Worker]'s [coroutineDispatcher] is overriden with a non-nullable
+ * [CoroutineDispatcher]. The new resulting binding dispatcher (e.g.RibDispatchers.Default) from
+ * [Worker] will take priority over the one passed via a [WorkerBinder.bind] call
  */
 public interface Worker {
 
   /**
-   * Specifies in which [CoroutineDispatcher] WorkerBind.bind will be operating on
+   * When overriden, will specify on which [CoroutineContext] the [Worker] will be bound via
+   * [WorkerBinder] (ignoring any [CoroutineDispatcher] being passed via [WorkerBinder])
    *
-   * NOTE: Default implementation will be using RibDispatchers.Unconfined to keep "caller" backward
-   * compatibility on existing usages
+   * For example given list of workers:
+   * 1) workers = listOf(defaultWorker, uiWorker)
+   * 2) Where [uiWorker] overrides [coroutineDispatcher] with [RibDispatchers.Main]
+   * 3) After calling WorkerBinder.bind(interactor, workers, RibDispatchers.IO). [uiWorker] will be
+   *    guaranteed to be called on RibDispatchers.Main
    */
   @JvmDefault
-  public val coroutineDispatcher: CoroutineDispatcher
-    get() = RibDispatchers.Unconfined
+  public val coroutineDispatcher: CoroutineDispatcher?
+    get() = null
```

Comment:
@psteiger Ended choosing a nullable dispatcher instead of EmptyCoroutineContext for easier Java support on existing Workers

Reply:
Out of curiosity, what was the issue with Java + EmptyCoroutineContext?

Reply:
I was getting an impossible cast from Dispatchers to CoroutineContext on java files. Let me double check the hierarchy to see what's going on

Reply:
Please ignore this, I was mistaken during local monorepo bump. This should always be a possible cast following. Will revert back